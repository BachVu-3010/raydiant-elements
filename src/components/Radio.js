import React from 'react';
import PropTypes from 'prop-types';
import FormControlLabel from './FormControlLabel';
import RadioInput from './RadioInput';


const propTypes = {
  /** Child elements are used as the checkbox label. */
  children: PropTypes.node,
  /** Whether the radio button is checked or not. */
  checked: PropTypes.bool,
  /** Whether the radio button is disabled or not. */
  disabled: PropTypes.bool,
  /** The group the radio button belongs to */
  name: PropTypes.string,
  /** The value of the radio button */
  value: PropTypes.string,
  /** Called when the user clicks the control. */
  onChange: PropTypes.func,
};
const defaultProps = {
  children: null,
  checked: false,
  disabled: false,
  name: '',
  value: '',
  onChange: () => {},
};

/**
 * Use this to select a value from several valid options.
 * See also: <a href="#checkbox">checkbox</a>, <a href="#select">select</a>.
 */
const Radio = ({ children, checked, disabled, name, onChange, value }) =>
  <FormControlLabel
    {...{ disabled }}
    control={<RadioInput {...{ checked, disabled, name, onChange, value }} />}
    label={children}
  />;


Radio.propTypes = propTypes;
Radio.defaultProps = defaultProps;

export default Radio;
