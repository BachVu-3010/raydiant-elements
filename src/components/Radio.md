Static:
```
<div>
  <Radio checked>I'm checked</Radio>
  <Radio checked={true}>I'm checked too</Radio>
  <Radio checked={false}>I'm not checked</Radio>
  <Radio>I'm not checked either</Radio>
</div>
```

Toggle:
```
initialState = {selected: "1"};
const onChange = (e) => { setState({selected: e.target.value}); };
<div>
  <Radio value="1" checked={state.selected === "1"} name="toggle" onChange={onChange}>Option 1</Radio>
  <Radio value="2" checked={state.selected === "2"} name="toggle" onChange={onChange}>Option 2</Radio>
</div>
```

Disabled:
```
<Radio checked disabled>Clicking on me won't do anything</Radio>
```