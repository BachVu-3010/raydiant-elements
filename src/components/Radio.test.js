import Radio from './Radio';
import toggleTest from '../test_helpers/toggle.test';

toggleTest(Radio);
