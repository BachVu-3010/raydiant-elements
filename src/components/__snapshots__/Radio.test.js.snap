// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<label
  className="MuiFormControlLabel-root-1 MuiFormControlLabel-hasLabel-3"
>
  <span
    className="MuiButtonBase-root-22 MuiIconButton-root-13 MuiSwitchBase-root-8 MuiSwitchBase-default-10 MuiRadio-default-5 MuiSwitchBase-checked-11 MuiRadio-checked-6"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role={undefined}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-19"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-24"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        checked={true}
        className="MuiSwitchBase-input-9"
        disabled={false}
        name=""
        onChange={[Function]}
        tabIndex={undefined}
        type="radio"
        value=""
      />
    </span>
    <span
      className="MuiTouchRipple-root-25"
    />
  </span>
  <p
    className="MuiTypography-root-32 MuiTypography-body1-41 MuiFormControlLabel-label-4"
  >
    Yes or no
  </p>
</label>
`;
